ucrop/src/main/java/com/yalantis/ucrop/task/BitmapCropTask.java

```diff
@@ -149,19 +151,36 @@ private boolean crop() throws IOException {
         Log.i(TAG, "Should crop: " + shouldCrop);
 
         if (shouldCrop) {
-            ExifInterface originalExif = new ExifInterface(mImageInputPath);
             saveImage(Bitmap.createBitmap(mViewBitmap, cropOffsetX, cropOffsetY, mCroppedImageWidth, mCroppedImageHeight));
             if (mCompressFormat.equals(Bitmap.CompressFormat.JPEG)) {
-                ImageHeaderParser.copyExif(originalExif, mCroppedImageWidth, mCroppedImageHeight, mImageOutputPath);
+                if (mImageInputUri != null && "content".equals(mImageInputUri.getScheme()) && "content".equals(mImageOutputUri.getScheme())) {
```

Comment:
Seems that null check for the mImageOutputUri is missed. And to make this lines a bit clearer, it'll be helpful to extract `"content"` as constant and reuse it for sure that there is no typo further. As an option, null-check with scheme check could be extracted to a method (to follow DRY principal and avoid mistakes while reuse)

Reply:
I agree with almost all those changes but I think we should take a more in depth look at this null check because that is a flaw that already existed before this PR. In fact `mImageOutputPath` can be null and can cause a `NullPointerException` in `ExifInterface` constructor.
I think booth `mImageOutputPath` and `mImageOutputUri` should be marked as `@NonNull` and they should never get all way down there as null. The `BitmapCropTask.crop` method itself should never be called if those variables are null because `BitmapCropTask.saveImage` depends on it.
I think that the only reason that those variables can be null is because the method `TransformImageView.setImageUri(@NonNull Uri imageUri, @Nullable Uri outputUri)` is been called on the `ResultActivitiy` of the sample project and it passes null to this. The sample project should not affect the architecture of the lib just to not repeat some coding.

Note: I may take some time to interact in this thread since I have to do it when I have time to do this and to analyse code may take more than just a few minutes, ok?

Reply:
@DeMoss15  I'm waiting a return about my considerations on null check before doing some change on the code, ok? Please let me know what do you think about it.

Reply:
Got your point, your'e right. But could you add this check here? Cause I don't sure when we will be able to change template and library.

Reply:
Fair enough but, since `mImageOutputUri` is now mandatory, I will add the protection on the `AsyncTask` by testing it on `doInBackground(Void... params)` where the error message can be correctly handled by uCrop default error handling mechanism through `mCropCallback.onCropFailure(t)` on `onPostExecute(@Nullable Throwable t)`. And latter, in another time, when finally the sample app gets separated from the lib code, `mImageOutputUri` can be marked as `@NonNull` on the rest of the code without any change on `BitmapCropTask` (If this class already exists the way it is now on this occasion since `AsyncTask` is now deprecated).